home/models.py

```diff
@@ -1222,3 +1214,35 @@ def clean(self):
 
     def __str__(self):
         return f'{self.locale}'
+
+
+def get_all_renditions_urls(image):
+    return [
+        rendition.url
+        for rendition
+        in image.get_rendition_model().objects.filter(image_id=image.id)
+    ]
+
+
+def extract_urls_from_rich_text(text):
+    return [
+        img['src']
+        for img
+        in BeautifulSoup(str(text), 'lxml').find_all('img')
+    ]
+
+
+def collect_urls_from_streamfield(field):
+    urls = []
+    for block in field:
```

Comment:
my preference would be to traverse all the blocks and gather all the image ids and at the end pass it to `get_all_renditions_urls` as a list to reduce the number of DB queries.

Reply:
I don't want to make a premature performance optimisation. This code is doing approximately the same as it was before, and I'm not sure if there was a performance issue before, but if there was, then perhaps it should be addressed separately.

Using a [renditions cache](https://docs.wagtail.org/en/stable/advanced_topics/performance.html#caching-image-renditions) would probably solve any performance issue, and we could set that up without Redis i.e. use the default in-memory cache, if we wanted to.